src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,60 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+import org.littletonrobotics.junction.Logger;
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.targeting.PhotonPipelineResult;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class Vision extends SubsystemBase {
+    private final PhotonCamera photonCamera;    
+    private final PhotonPoseEstimator poseEstimator;
+    private final BiConsumer<Pose2d, Double> consumer;
+
+    public Vision(BiConsumer<Pose2d, Double> consumer) throws IOException{
+        photonCamera = new PhotonCamera(Constants.VisionConstants.kCameraName);
+        poseEstimator = new PhotonPoseEstimator(
+            AprilTagFieldLayout.loadFromResource(AprilTagFields.k2024Crescendo.m_resourceFile),
+            PhotonPoseEstimator.PoseStrategy.MULTI_TAG_PNP_ON_RIO,
+            Constants.VisionConstants.kCameraOffset);
+        this.consumer = consumer;
```

Comment:
As we tested at the end of the meeting, we can improve the pose estimation performance by switching to the `MULTI_TAG_PNP_ON_COPROCESSOR` strategy and passing the `photonCamera` as the 3rd argument (before the camera offset).